utils/services/network/RtdbModules.ts

```diff
@@ -90,12 +94,17 @@ export async function rtdbTopicGet(): Promise<string[]> {
   const path = `topicList`;
   // const query = child(ref(db),path);
   const rr = ref(db, path);
-  const qq = query(rr, orderByValue());
-  const res = await get(qq).then((snapshot)=>{
-    // console.log(snapshot);
-    if(snapshot.exists()){
-      // console.log(snapshot.val());
-      return snapshot.val();
+  const qq = query(rr, orderByChild("name"));
+  const res = await get(qq).then((snapshot) => {
+    if (snapshot.exists()) {
+      
+      const data: string[] = [];
+      // using `forEach` because normal `val` won't guarantee the ordering request
+      snapshot.forEach((e) => {
```

Comment:
@arunstop So basically we cannot use just play `.val()` from the snapshot. Cuz it said in the documentation (when i hover on my IDE) that `.val()` doesn't guarantee the requested order. The documentation also said exactly why you should use `.forEach()` instead, cuz it guarantees the requested order.


Pretty weird technicality i would say, but i guess it must have been designed exactly like that for performance reason or something. After all they are Google engineers who alchemized all of this! I was so stunted, weirded and all other negative feeling when i was struggling with this.

The bright side is #TodayILearned :tada:

Reply:
Alright! Today i definitely learned as well then! Thanks for the info! :+1: